storage/store/proposal_signatures.go

```diff
@@ -35,13 +37,13 @@ func newProposalSignatures(collector module.CacheMetrics, db storage.DB) *propos
 		db: db,
 		cache: newCache(collector, metrics.ResourceProposalSignature,
 			withLimit[flow.Identifier, []byte](4*flow.DefaultTransactionExpiry),
-			withStore(store),
+			withStoreWithLock(store),
 			withRetrieve(retrieve)),
 	}
 }
 
-func (h *proposalSignatures) storeTx(rw storage.ReaderBatchWriter, blockID flow.Identifier, sig []byte) error {
-	return h.cache.PutTx(rw, blockID, sig)
+func (h *proposalSignatures) storeTx(lctx lockctx.Proof, rw storage.ReaderBatchWriter, blockID flow.Identifier, sig []byte) error {
```

Comment:
lets clearly document which lock is required and the expectations on the caller:
```suggestion
// storeTx persists the given `sig` as the proposer's signature for the specified block.
//
// CAUTION:
//   - The caller must acquire either the lock [storage.LockInsertBlock] or [storage.LockInsertOrFinalizeClusterBlock]
//     but not both and hold the lock until the database write has been committed.
//   - OVERWRITES existing data (potential for data corruption):
//     The lock proof serves as a reminder that the CALLER is responsible to ensure that the DEDUPLICATION CHECK
//     is done elsewhere ATOMICALLY with this write operation. It is intended that this function is called only for new
//     blocks, i.e. no signature was previously persisted for it.
//
// No error returns expected during normal operations.
func (h *proposalSignatures) storeTx(lctx lockctx.Proof, rw storage.ReaderBatchWriter, blockID flow.Identifier, sig []byte) error {
```